rs/replica/setup_ic_network/src/lib.rs

```diff
@@ -346,7 +346,14 @@ fn start_consensus(
 
         join_handles.push(jh);
 
-        new_p2p_consensus.add_client(consensus_rx, consensus_pool, consensus_gossip, client);
+        let downloader = ic_artifact_downloader::FetchArtifact::new(
+            log.clone(),
+            rt_handle.clone(),
+            consensus_pool,
+            consensus_gossip,
+            metrics_registry.clone(),
+        );
```

Comment:
Can't we have this as the default download in the consensus manager builder, and have a `with_downloader` function if we want to inject specific download logic?

Reply:
It would avoid much of the duplication.

Reply:
The reason I didn't do this was to avoid a dependency between `ic-artifact-downloader` and `ic-consensus-manager`.